src/lib/redux/siteDataSlice.ts

```diff
@@ -69,9 +92,15 @@ const siteDataSlice = createSlice({
         ]
         }
       }
-      }
+    },
+    editCustomer(state, action) {
+      const stateCopy = JSON.parse(JSON.stringify(state));
+      const index = stateCopy.currentSite.customers.findIndex((e: CustomerRecord) => e.id === action.payload.id);
```

Comment:
I think this might miss if the customer was created offline --> edited offline. Let's talk about how that should work offline, because we can handle it the same way we handle payments and meter readings